### PR TITLE
[Issue 174] ADR and coverage threshold

### LINKED ISSUE #174: [ADR]: Backend Testing Framework and Coverage
### Description

This ADR will record the decision for a backend testing framework and coverage.

Things to keep in mind are testing types (functional, unit etc), and the Python tools to utilize, code coverage, and other metrics.

### Options

- pytest

### Decision Criteria

_No response_

### Definition of Done

- [ ] The ADR is drafted and stored in `documentation/decisions/adr`
- [ ] The ADR is included in the necessary table of contents in the ADR/decision folder
- [ ] The ADR describes the drivers and pros and cons of the decision
- [ ] The ADR has been reviewed and approved by the internal team
- [ ] The ADR satisfies requirements that are outlined in the ADR template

### PR DESCRIPTION
**Pull request by SammySteiner**
## Summary
Fixes #174

### CODE REVIEW

File: documentation/decisions/adr/0013-backend-testing.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+### Pros and Cons of the Options <!-- OPTIONAL -->
+#### [Pytest](https://docs.pytest.org/)
+The pytest framework makes it easy to write small, readable tests, and can scale to support complex functional testing for applications and libraries.
+
+
+- **Pros**
+  - Lightweight, well-supported and documented testing solution
+  - Already integrated in Flask back-end template
+  - Modular fixtures for managing small or parametrized long-lived test resources
+  - Can run unittest (including trial) and nose test suites out of the box
```
Comment (daphnegold) [inline documentation/decisions/adr/0013-backend-testing.md:37]:
```suggestion
  - Can run [unittest](https://docs.python.org/3/library/unittest.html) (including trial) and nose test suites out of the box
```

File: documentation/decisions/adr/0013-backend-testing.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+## Decision Outcome <!-- REQUIRED -->
+
+### Unit Testing
+**Pytest**, because it is integrated into the Nava Flask template application, well-maintained, and lightweight. Importantly, documentation is thorough and helpful information for troubleshooting can be easily accessed.
+
+### Testing Coverage
+**Coverage**, because it is integrated into the Nava Flask template application, well-maintained, and lightweight. Importantly, documentation is thorough and helpful information for troubleshooting can be easily accessed.
+
+We added a code coverage threshold of %80 in `api/pyproject.toml`
```
Comment (daphnegold) [inline documentation/decisions/adr/0013-backend-testing.md:59]: 🥳

**Review by daphnegold — APPROVED** [review]
:shipit:

**Review by widal001 — APPROVED** [review]
LGTM!! Yay for `pytest`!!!

**Pull request merged**